### ISSUE TITLE
Bug: Incomplete CSS margin parsing if "auto" or "0" without unit is used

### ISSUE THREAD

**Issue by sserdyuk** [ckeditor/ckeditor4#1330]
## Are you reporting a feature request or a bug?

Bug

Related to #450 

Open HTML with inline CSS like `margin: 10px 0 0;` switch to Source mode, or
open HTML with inline CSS like `margin: 10px auto 0;` switch to Source mode.

### Expected result

`margin-bottom:0; margin-left:0; margin-right:0; margin-top:10px`, or
`margin-bottom:0; margin-left:auto; margin-right:auto; margin-top:10px`

### Actual result

`margin-bottom:10px; margin-left:10px; margin-right:10px; margin-top:10px`

*What is the actual result of the above steps?*

Incorrect margins are applied

**Comment by engineering-this** [comment]
I just found that problem when working with PWT.

There are more cases of buggy margins:

Case 1:
```//Input:
<p style="margin: 0 0 12px 0">ASD</p>
//Expected:
<p style="margin-bottom: 12px">ASD</p>
//Actual:
<p style="margin-left:12px; margin-right:12px">ASD</p>```

Case 2:
```
//Input:
<p style="margin: 0 12px 6px">ASD</p>
//Expected:
<p style="margin-left: 12px; margin-bottom: 6px; margin-right: 12px">ASD</p>
//Actual
<p style="margin-left:6px; margin-right:6px">ASD</p>
```

And probably many other cases.

**Comment by achrysanthakopoulou** [comment]
Is this bug going to be fixed ?
It seems that I didn't have this issue until I used this code in my config:

	config.allowedContent = {
	    $1: {
	        elements: CKEDITOR.dtd,
	        attributes: true,
                styles: true,
	        classes: true
	    }
	};

**Comment by achrysanthakopoulou** [comment]
Anyone?

**Comment by f1ames** [comment]
Hello @achrysanthakopoulou,
I have added this issue to our Backlog, however as we have other priorities at the moment, there is no ETA for this one.

For anyone interested in fixing this issue, please add 👍 reaction to the issue report.

**Comment by achrysanthakopoulou** [comment]
Hello @f1ames 
I tried with a colleague of mine to fix the issue.

Please check the following code:
Replace this regex at production's ckeditor.js main file: 
.match(/(\-?[\.\d]+\w+)/g)
with this:
.match(/(\-?[\.\d]*\w+)/g)

It solves my problem but needs testing.

At ckeditor-dev you can find it in :
https://github.com/ckeditor/ckeditor-dev/blob/606ced7/core/filter.js#L2201

**Comment by f1ames** [comment]
@achrysanthakopoulou thanks for checking this issue!

Maybe you would like to prepare a PR with the fix (please see [contributing guide](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html))?

**Comment by f1ames** [comment]
Fixed in #4270.
